### DISCUSSION TITLE
How to group data by month?

### DISCUSSION THREAD

**Discussion by odysseyskate** [Budibase/budibase#3170] [Help] [answered]
I want to have a bar chart that shows total amounts by month.

Ideally it would show all 12 months, even if there's no data for some of them.

Is this possible?

The closest I could get is in the data source to calculate total amount & then group by date but this seems to group by the exact date only.

Thanks

**Comment by mjashanks** [comment, marked as answer]
Unfortunately, this is not possible right now. The solution would be to allow grouping by a formula column, which is recorded here: https://github.com/Budibase/budibase/discussions/2999

I have linked to this discussion from there!

**Comment by datdamnzotz** [comment]
Also depending on your data source you might be able to do this in a custom query
MySQL by date example:
```
SELECT DATE_FORMAT(logtime,'%Y-%m-%d') as Date, SUM(Counts) as Count
FROM `log`
GROUP BY YEAR(logtime), MONTH(logtime), DAY(logtime)
ORDER BY Date DESC
```
or by month
```
SELECT DATE_FORMAT(logtime,'%Y-%m') as Date, SUM(Counts) as Count
FROM `log`
GROUP BY YEAR(logtime), MONTH(logtime)
ORDER BY Date DESC
```
